src/VisualStudio/Core/Impl/Options/Style/NamingPreferences/NamingStyleViewModel.cs

```diff
@@ -0,0 +1,180 @@
+using Microsoft.CodeAnalysis.Notification;
+using Microsoft.VisualStudio.LanguageServices.Implementation.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options.Style.NamingPreferences
+{
+    internal class NamingStyleViewModel : AbstractNotifyPropertyChanged
+    {
+        private NamingStyle _style;
+
+        public NamingStyleViewModel(NamingStyle style, INotificationService notificationService)
+        {
+            _notificationService = notificationService;
+            _style = style;
+            this.ID = style.ID;
+            this.RequiredPrefix = style.Prefix;
+            this.RequiredSuffix = style.Suffix;
+            this.WordSeparator = style.WordSeparator;
+            this.FirstWordGroupCapitalization = (int)style.CapitalizationScheme;
+            this.NamingConventionName = style.Name;
+
+            CapitalizationSchemes = new List<CapitalizationDisplay>
+                {
+                    new CapitalizationDisplay(Capitalization.PascalCase, ServicesVSResources.CapitalizationStyleExample_PascalCase),
+                    new CapitalizationDisplay(Capitalization.CamelCase, ServicesVSResources.CapitalizationStyleExample_CamelCase),
+                    new CapitalizationDisplay(Capitalization.FirstUpper, ServicesVSResources.CapitalizationStyleExample_FirstWordUpper),
+                    new CapitalizationDisplay(Capitalization.AllUpper, ServicesVSResources.CapitalizationStyleExample_AllUpper),
+                    new CapitalizationDisplay(Capitalization.AllLower, ServicesVSResources.CapitalizationStyleExample_AllLower)
+                };
+
+            CapitalizationSchemeIndex = CapitalizationSchemes.IndexOf(CapitalizationSchemes.Single(s => s.Capitalization == style.CapitalizationScheme));
+        }
+
+        public IList<CapitalizationDisplay> CapitalizationSchemes { get; set; }
+
+        private int _capitalizationSchemeIndex;
+        public int CapitalizationSchemeIndex
+        {
+            get
+            {
+                return _capitalizationSchemeIndex;
+            }
+            set
+            {
+                _style.CapitalizationScheme = CapitalizationSchemes[value].Capitalization;
+                if (SetProperty(ref _capitalizationSchemeIndex, value))
+                {
+                    NotifyPropertyChanged(nameof(CurrentConfiguration));
+                }
+            }
+        }
+
+        public Guid ID { get; internal set; }
+
+        private string _namingConventionName;
+        public string NamingConventionName
+        {
+            get { return _namingConventionName; }
+            set { SetProperty(ref _namingConventionName, value); }
+        }
+
+        public string CurrentConfiguration
+        {
+            get
+            {
+                return _style.CreateName(new[] { "test", "variable", "name" });
+            }
+            set
+            {
+            }
+        }
```

Comment:
the getter and setter look odd here. was this test code forgotten to be removed?
its not loc 'd but i expect you don't even want this perhaps?